File: src/com/xw/vvtts/core/VvttsCore.java
package com.xw.vvtts.core;

/**
 * 自研 ECI 桥接（libvvtts_core.so）v3
 * 中英文各一个 session（dialect handle），角色由反引号 annotation 驱动。
 */
public class VvttsCore {

    static {
        System.loadLibrary("vvtts_core");
    }

    public static final int CHARSET_1252 = 0;   // Windows-1252（西文）
    public static final int CHARSET_GBK = 6;    // 中文 GB18030

    static native long nativeInitEngine(String configDir, String libDir, int dialect);
    static native short[] nativeSynthesize(long handle, int dialect, byte[] text, int charsetId, String outPath);
    static native int nativeSetVoiceParam(long handle, int voice, int param, int value);
    static native int nativeGetVoiceParam(long handle, int voice, int param);
    static native int nativeSetParam(long handle, int param, int value);
    static native int nativeSetStandardVoice(long handle, int voiceNumber);
    static native void nativeShutdown(long handle);
    static native void nativeStop(long handle);

    /** 为指定 dialect 创建 session（重复调用会重建） */
    public static long openEngine(String configDir, String libDir, int dialect) {
        try {
            return nativeInitEngine(configDir, libDir, dialect);
        } catch (Throwable e) {
            android.util.Log.e("VvttsCore", "openEngine native crash", e);
            return 0;
        }
    }

    /** 合成。textBytes 已编码。返回 PCM short 数组。 */
    public static short[] synth(long handle, int dialect, byte[] text, int charsetId, String outPath) {
        try {
            return nativeSynthesize(handle, dialect, text, charsetId, outPath);
        } catch (Throwable e) {
            android.util.Log.e("VvttsCore", "synth native crash", e);
            return null;
        }
    }

    /** ECI 语音参数（voice=0 当前活动语音） */
    public static int setVoiceParam(long handle, int voice, int param, int value) {
        try {
            return nativeSetVoiceParam(handle, voice, param, value);
        } catch (Throwable e) {
            return -1;
        }
    }

    public static int getVoiceParam(long handle, int voice, int param) {
        try {
            return nativeGetVoiceParam(handle, voice, param);
        } catch (Throwable e) {
            return -1;
        }
    }

    /** 引擎参数（2=音调 5=音量 7=语速） */
    public static int setParam(long handle, int param, int value) {
        try {
            return nativeSetParam(handle, param, value);
        } catch (Throwable e) {
            return -1;
        }
    }

    /** 切换标准 voice */
    public static int setStandardVoice(long handle, int voiceNumber) {
        try {
            return nativeSetStandardVoice(handle, voiceNumber);
        } catch (Throwable e) {
            return -1;
        }
    }

    public static void shutdown(long handle) {
        try {
            nativeShutdown(handle);
        } catch (Throwable e) {}
    }

    public static void stop(long handle) {
        try {
            nativeStop(handle);
        } catch (Throwable e) {}
    }
}

File: src/com/xw/vvtts/core/VvttsCore.kt
package com.xw.vvtts.core

import android.util.Log

/**
 * In-house ECI bridge (libvvtts_core.so)v3
 * One session per language(a dialect handle);voices are driven by backtick annotations.
 *
 * All native declarations must be @JvmStatic(static methods land on the outer class),
 * 否则 Kotlin 会把它编译进 Companion,JNI 符号名会变成
 * Java_com_xw_vvtts_core_VvttsCore_00024Companion_native*,C 桥接就找不到方法了.
 */
class VvttsCore {

    companion object {
        const val CHARSET_1252 = 0   // Windows-1252 (Western
        const val CHARSET_GBK = 6    // Chinese GB18030

        init {
            System.loadLibrary("vvtts_core")
        }

        @JvmStatic external fun nativeInitEngine(configDir: String, libDir: String, dialect: Int): Long
        @JvmStatic external fun nativeSynthesize(handle: Long, dialect: Int, text: ByteArray, charsetId: Int, outPath: String?): ShortArray?
        @JvmStatic external fun nativeSetVoiceParam(handle: Long, voice: Int, param: Int, value: Int): Int
        @JvmStatic external fun nativeGetVoiceParam(handle: Long, voice: Int, param: Int): Int
        @JvmStatic external fun nativeSetParam(handle: Long, param: Int, value: Int): Int
        @JvmStatic external fun nativeSetStandardVoice(handle: Long, voiceNumber: Int): Int
        @JvmStatic external fun nativeShutdown(handle: Long)
        @JvmStatic external fun nativeStop(handle: Long)

       /** Create a session for the given dialect(repeated calls rebuild it) */
        @JvmStatic
        fun openEngine(configDir: String?, libDir: String?, dialect: Int): Long {
            return try {
                nativeInitEngine(configDir!!, libDir!!, dialect)
            } catch (e: Throwable) {
                Log.e("VvttsCore", "openEngine native crash", e)
                0
            }
        }

       /** Synthesize.textBytes is pre-encoded.Returns a PCM short array. */
        @JvmStatic
        fun synth(handle: Long, dialect: Int, text: ByteArray, charsetId: Int, outPath: String?): ShortArray? {
            return try {
                nativeSynthesize(handle, dialect, text, charsetId, outPath)
            } catch (e: Throwable) {
                Log.e("VvttsCore", "synth native crash", e)
                null
            }
        }

       /** ECI voice params(voice=0 is the active voice) */
        @JvmStatic
        fun setVoiceParam(handle: Long, voice: Int, param: Int, value: Int): Int {
            return try {
                nativeSetVoiceParam(handle, voice, param, value)
            } catch (e: Throwable) {
                -1
            }
        }

        @JvmStatic
        fun getVoiceParam(handle: Long, voice: Int, param: Int): Int {
            return try {
                nativeGetVoiceParam(handle, voice, param)
            } catch (e: Throwable) {
                -1
            }
        }

       /** Engine params(2=pitch 5=volume 7=speed) */
        @JvmStatic
        fun setParam(handle: Long, param: Int, value: Int): Int {
            return try {
                nativeSetParam(handle, param, value)
            } catch (e: Throwable) {
                -1
            }
        }

       /** Switch to a standard voice */
        @JvmStatic
        fun setStandardVoice(handle: Long, voiceNumber: Int): Int {
            return try {
                nativeSetStandardVoice(handle, voiceNumber)
            } catch (e: Throwable) {
                -1
            }
        }

        @JvmStatic
        fun shutdown(handle: Long) {
            try {
                nativeShutdown(handle)
            } catch (e: Throwable) {
                // ignore
            }
        }

        @JvmStatic
        fun stop(handle: Long) {
            try {
                nativeStop(handle)
            } catch (e: Throwable) {
                // ignore
            }
        }
    }

       /** Stateless bridge class(default constructor kept for compat with old new VvttsCore() call sites) */
}